ledger/acctupdates_test.go

```diff
@@ -184,16 +188,24 @@ func (ml *mockLedgerForTracker) Close() {
 }
 
 func (ml *mockLedgerForTracker) Latest() basics.Round {
+	ml.mu.RLock()
```

Comment:
Actually, several tests fail with race detection: `TestFirstStagePersistence`, `TestSecondStagePersistence`, and more.

The race started because now `votersTracker` implements a `postCommit` function. In `postCommit` function it accesses `blockHdr` from the ledger.

i.e due to this change the commit process accesses blockhdr

Reply:
> TestFirstStagePersistence...

weird, the test does not close the ledger to race on state. Can we look deeper into the race reason?

Reply:
When initialized, the catchpoint tracker starts a goroutine: [`go tr.commitSyncer(tr.deferredCommits)`](ledger/tracker.go:307) which requests block headers from the ledger (`mockLedger` in our case) periodically on `postCommit`.

Meanwhile, the main goroutine running the test is adding blocks to the `mockLedger` using `addMockBlock`.
This is the potential data race and it is apparent in every unit test using the `catchpointTracker`.